sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/FileFormatWriter.scala

```diff
@@ -314,21 +339,40 @@ object FileFormatWriter extends Logging {
 
           recordsInFile = 0
           releaseResources()
+          numOutputRows += recordsInFile
           newOutputWriter(fileCounter)
         }
 
         val internalRow = iter.next()
+        val startTime = System.nanoTime()
         currentWriter.write(internalRow)
+        timeOnCurrentFile += (System.nanoTime() - startTime)
```

Comment:
I'm worried about this per-row time tracking may have performance penalty, and also may be inaccurate because the real data flow is pipelined. cc @rxin who should have more experience in this area.